src/main/java/core/basesyntax/model/zoo/Animal.java

```diff
@@ -1,9 +1,33 @@
 package core.basesyntax.model.zoo;
 
+import javax.persistence.Column;
+import javax.persistence.DiscriminatorColumn;
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.Inheritance;
+import javax.persistence.InheritanceType;
+
+@Entity
+@Inheritance(strategy = InheritanceType.SINGLE_TABLE)
+@DiscriminatorColumn(name = "dtype")
 public class Animal {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    @Column(name = "id", updatable = false, nullable = false)
```

Comment:
PK can't be null by default
I think this `@Column` annotation here is redundant

Reply:
I'll remove it but this is for id constraints, that you can't update id and id shouldn't be the null